src/libraries/Common/src/Interop/Windows/Netapi32/Interop.DsRoleGetPrimaryDomainInformation.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Netapi32
+    {
+        internal enum DSROLE_PRIMARY_DOMAIN_INFO_LEVEL
+        {
+            DsRolePrimaryDomainInfoBasic = 1,
+            DsRoleUpgradeStatus = 2,
+            DsRoleOperationState = 3,
+            DsRolePrimaryDomainInfoBasicEx = 4
+        }
+
+        [LibraryImport(Libraries.Netapi32, StringMarshalling = StringMarshalling.Utf16)]
+        internal static partial int DsRoleGetPrimaryDomainInformation(
+            string lpServer,
+            DSROLE_PRIMARY_DOMAIN_INFO_LEVEL InfoLevel,
+            out IntPtr Buffer);
+
+        [LibraryImport(Libraries.Netapi32, StringMarshalling = StringMarshalling.Utf16)]
+        internal static partial int DsRoleGetPrimaryDomainInformation(
+            IntPtr lpServer,
+            DSROLE_PRIMARY_DOMAIN_INFO_LEVEL InfoLevel,
+            out IntPtr Buffer);
```

Comment:
The two overloads can be consolidated into one overload with `string?`. `IntPtr.Zero` at call site should be replaced to `null`.

Reply:
Nice, done in c7d261a55b2f9ea572a3c3541bc4f9cda4815ca6